src/shared/config/conf.cpp

```diff
@@ -0,0 +1,294 @@
+      create();
+      load();
+   }
+
+   print_table(m_conf_table);
+}
+
+Config& Config::get_instance() {
+   
+   static Config m_instance;
+
+   return m_instance;
+}
+
+bool Config::create(){
+
+   LOG_INFO("Creating skype directory and config file");
+   
+   bool res = fs::is_directory(m_user_dir);
+   if ( false == res ){
+      res = create_conf_directory();
+   }
+   
+   if ( false == res ){
+      return is_valid(NEG, "Couldn't create directory", Error);
+   }
+ 
+   copy_conf();
+
+   return res;
+}
+
+bool Config::load(){
+
+   std::ifstream file(m_user_conf);
+   std::string line;
+   
+   int res = file.good();
+   if ( true == res){
+
+      LOG_INFO("Parsing config file");
+
+      while (std::getline(file,line)){
+         
+         res = append_to_map(line);
+         if ( false == res){
+            return is_valid(NEG, "Parsing the config file failed!", Error);
+         }
+
+      }
+      
+      file.close();
+   }
+   else 
+   {
+      return is_valid(NEG, "Couldn't open file", Error);
+   }
+
+
+   return true;  
+}
+
+bool Config::save(){
+
+   std::ofstream user_file;
+   user_file.open(m_user_conf, std::ios::out | std::ofstream::trunc | std::ios::binary);
+   
+   std::string buffer = "";
+
+   for ( auto it : m_conf_table ){
+     user_file << to_string(it.first) + "=" + it.second + "\n";
+   }
+
+   user_file.close();
+
+   return true;
+}
+
+bool Config::update_variable(Configuration t_variable, std::string t_value) {
+
+   bool res = (t_variable == SERVER_IP && t_variable < LAST) ? true : false;
+
+   if ( false == res || t_value.empty() ) {
+      return is_valid(NEG, "Couldn't update variable", Error);
+   }
+
+   m_conf_table.at(t_variable) = t_value;
+
+   return true;
+}
+
+bool Config::corrupted_conf() {
+
+   LOG_ERR("Corrupted config file, fixing it");
+   
+   bool res = create();
+   if ( false == res ){
+      return is_valid(NEG, "Could'nt fix config file", Error);
+   }
+
+   return true;
+}
+
+bool Config::corrupted_variable(Configuration t_variable){
+
+   std::ifstream file(m_base_conf);
+   std::string line;
+   
+   int res = file.good();
+   if ( true == res){
+
+      while (std::getline(file,line)){
+
+         std::string variable = parse_line(line);
+         if ( variable == to_string(t_variable)) {
+            
+            m_conf_table.at(t_variable) = line;
+         }         
+      }
+      
+      file.close();
+   }
+   else 
+   {
+      return is_valid(NEG, "Couldn't open file", Error);
+   }
+
+
+   return true; 
+
+}
+/* Private */
+
+bool Config::compare_files(){
+
+   std::ifstream base(m_base_conf, std::ifstream::ate | std::ifstream::binary);
+   std::ifstream user(m_user_conf, std::ifstream::ate | std::ifstream::binary);
+   
+   const std::ifstream::pos_type base_size = base.tellg();
+   const std::ifstream::pos_type user_size = user.tellg();
+
+   if (user_size != base_size ){
+
+      is_valid(NEG, "Config files have different sizes", Error);
```

Comment:
can you change the `is_valid` method so it fits the validation needs of this class?  This way you wouldn't have to use `NEG` to make the method usable here.